freepass project - anizone/src/App.js

```diff
@@ -0,0 +1,72 @@
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import Footer from './components/Footer';
+import Loading from './components/Loading';
+import Navbar from './components/Navbar';
+import GlobalProvider from './context/GlobalContext';
+import GenreProvider from './context/GenreContext';
+import AnimeDetail from './pages/AnimeDetail';
+import CharacterDetail from './pages/CharacterDetail';
+import CurrentAnime from './pages/CurrentAnime';
+import Genres from './pages/Genres';
+import Home from './pages/Home';
+import SearchAnime from './pages/SearchAnime';
+import TopAnime from './pages/TopAnime';
+import UpcomingAnime from './pages/UpcomingAnime';
+import TopAnimeProvider from './context/TopAnimeContext';
+import OngoingProvider from './context/OngoingContext';
+import UpcomingProvider from './context/UpcomingContext';
```

Comment:
I think you don't need these 3 contexts, because you're only using them on their own page and this will reduce the web performance because it will make the context be called even on the page that doesn't use it. You only use context if there are functions or data that are needed across pages or components, just like the GlobalContext you made.

The main goal of using context is to avoid prop drilling, so it would be better if you just use a simple useState, useReducer, or maybe a simple js function to replace these 3 contexts.

Reply:
Thanks for the feedbacks :)